### PR TITLE
Add pre-commit utilities

### PR DESCRIPTION
**Pull request by vwxyzjn**
This PR introduces pre-commit pipelines, which include

* black: auto-format
* codespell: check word spelling
* mypy

going to add the following hooks in a follow-up PR:

* isort: automatically sort imported dependencies
* autoflake: automatically remove unused variables and unused imports

However, it does keep the tests alone because they run slow: see https://github.com/pre-commit/pre-commit-hooks/issues/291

### CODE REVIEW

**Comment by vwxyzjn** [comment]
Hey, @cswinter could you remove the black and mypy as the required CIs?

image

File: .github/workflows/checks.yaml [hunk 1]
```diff
@@ -62,72 +62,72 @@ jobs:
...
-          poetry run mypy enn_zoo
-          poetry run mypy entity_gym
-          poetry run mypy rogue_net
-          poetry run mypy enn_ppo
+  #     - name: MyPy
+  #       run: |
+  #         poetry run mypy enn_zoo
+  #         poetry run mypy entity_gym
+  #         poetry run mypy rogue_net
+  #         poetry run mypy enn_ppo
```
Comment (cswinter) [inline .github/workflows/checks.yaml:130]: remove commented out config

**Review by cswinter — COMMENTED** [review]
Could you add some instructions for how to set up pre-commit locally?

**Comment by vwxyzjn** [comment]
Merging this now.

**Pull request merged**